### ISSUE TITLE
When deleting a karmada instance, some pods (metrics-adapters) are still running.

### ISSUE THREAD

**Issue by sykim-etri** [karmada-io/karmada#4030]
**What happened**:
I deleted a karmada instance, BUT only metics-adapter pods are still running.

**What you expected to happen**:
When deleting a karmada instance, all pods related the karmada instance should be terminated. (is it right?)

**How to reproduce it (as minimally and precisely as possible)**:
I followed this guide (https://github.com/karmada-io/karmada/blob/master/operator/README.md).

I run this script.
```
kubectl apply -f - <<EOF
apiVersion: operator.karmada.io/v1alpha1
kind: Karmada
metadata:
  name: karmada-demo
  namespace: test
EOF
```

And then I could check the karmada's pods as follows:
```
NAME                                                    READY   STATUS    RESTARTS   AGE                                                                                                                         
karmada-demo-aggregated-apiserver-cd8c685bb-2d8s5       1/1     Running   0          53s
karmada-demo-apiserver-6c74757b8b-54jnk                 1/1     Running   0          74s
karmada-demo-controller-manager-5cb655fc64-8kj8p        1/1     Running   0          48s
karmada-demo-etcd-0                                     1/1     Running   0          75s
karmada-demo-kube-controller-manager-68f5649c7b-7rnjt   1/1     Running   0          48s
karmada-demo-metrics-adapter-65cc4546d9-25crs           1/1     Running   0          48s
karmada-demo-metrics-adapter-65cc4546d9-x5r25           1/1     Running   0          48s
karmada-demo-scheduler-7cddd77f9-5p4mk                  1/1     Running   0          48s
karmada-demo-webhook-67d895c879-zdlzp                   1/1     Running   0          48s
```

And then I tried to delete the karmada instance.
```
$ kubectl delete karmada karmada-demo -n test
karmada.operator.karmada.io "karmada-demo" deleted
$
```

BUT I could see some pods in test namespace.
```
$ kubectl get pods -n test
NAME                                            READY   STATUS    RESTARTS   AGE
karmada-demo-metrics-adapter-65cc4546d9-25crs   1/1     Running   0          3m30s
karmada-demo-metrics-adapter-65cc4546d9-x5r25   1/1     Running   0          3m30s
```

**Anything else we need to know?**:

**Environment**:
- Karmada version: v1.8.0-alpha.0
- kubectl-karmada or karmadactl version (the result of `kubectl-karmada version` or `karmadactl version`):
```
$ karmadactl version
karmadactl version: version.Info{GitVersion:"v0.0.0-master", GitCommit:"unknown", GitTreeState:"unknown", BuildDate:"unknown", GoVersion:"go1.21.0", Compiler:"gc", Platform:"linux/amd64"}
```
- Others:

**Comment by chaunceyjiang** [comment]
/assign

I will try to fix this issue.

**Comment by RainbowMango** [comment]
@chaunceyjiang Do you have a clue about this?

**Comment by chaunceyjiang** [comment]
>  @chaunceyjiang Do you have a clue about this?

No, My guess is that it's because #3732 did not implement "deletion". Of course, this is just a speculation and I need to investigate further.

**Comment by RainbowMango** [comment]
@sykim-etri Thanks for spotting, This issue has been fixed and planned to backport to release-1.7. 
You can have a try against master now.
